IMDb.js

```diff
@@ -60,98 +64,78 @@ function getSearchResults(doc, checkOnly) {
 	return found ? items : false;
 }
 
-
 function doWeb(doc, url) {
 	if (detectWeb(doc, url) == "multiple") {
 		Zotero.selectItems(getSearchResults(doc, false), function (items) {
 			if (!items) {
-				return true;
+				return;
 			}
 			var articles = [];
 			for (var i in items) {
 				articles.push(i);
 			}
 			ZU.processDocuments(articles, scrape);
 		});
-	} else {
+	}
+	else {
 		scrape(doc, url);
 	}
 }
 
-
 function scrape(doc, url) {
 	var translator = Zotero.loadTranslator('web');
 	// Embedded Metadata
 	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
-	//translator.setDocument(doc);
+	// translator.setDocument(doc);
 	
 	translator.setHandler('itemDone', function (obj, item) {
-		var titleWrapper = ZU.xpath(doc, '//div[contains(@class, "title_wrapper")]');
-		var title = ZU.xpathText(titleWrapper, './h1/text()[1]');
-		if (title) {
-			item.title = title;
-		}
-		item.date = ZU.xpathText(titleWrapper, './/meta[@itemprop="datePublished"]/@content');
-		item.runningTime = ZU.xpathText(titleWrapper, './/time[@itemprop="duration"]');
-		item.genre = ZU.xpathText(titleWrapper, './/span[@itemprop="genre"]');
-		var origTitle = ZU.xpathText(titleWrapper, './/div[contains(@class, "originalTitle")]/text()[1]');
-		if (origTitle) {
-			addExtra(item, "original-title: "+origTitle);
-		}
-		var pageId = ZU.xpathText(doc, '//meta[@property="pageId"]/@content');
-		if (pageId) {
-			addExtra(item, "IMDb ID: "+pageId);
-		}
-		
-		var summary = ZU.xpath(doc, '//div[contains(@class, "plot_summary_wrapper")]');
+		let json = JSON.parse(text(doc, 'script[type="application/ld+json"]'));
+		item.title = json.name;// note that json only has the original title
+		item.date = "datePublished" in json ? json.datePublished : "";
+		item.runningTime = "duration" in json ? json.duration.replace("PT", "").toLowerCase() : "";
+		item.genre = "genre" in json ? json.genre : "";
+
+		item.abstractNote = "description" in json ? json.description : "";
 		var creatorsMapping = {
-			"director": "director",
-			"creator": "scriptwriter",
-			"actors": "contributor"
+			director: "director",
+			creator: "scriptwriter",
+			actor: "contributor"
 		};
 		for (var role in creatorsMapping) {
-			var creators = ZU.xpath(summary, './/span[@itemprop="'+role+'"]//span[@itemprop="name"]');
-			for (var i=0; i<creators.length; i++) {
-				item.creators.push(ZU.cleanAuthor(creators[i].textContent, creatorsMapping[role]));
+			if (!(role in json)) continue;
```

Comment:
For our purposes, there's basically no need for `in` anywhere. So, e.g., you can just use `if (!json.role) continue;`.